I did some more digging. I saw that the test in the CI fails with permission errors.

```
| No symbol table is loaded.  Use the "file" command.
| warning: opening /proc/PID/mem file for lwp 4103309.4103309 failed: Permission denied (13)
| Traceback (most recent call last):
|   File "/var/lib/buildkite-agent/builds/linux-56-59b8f5d88-h5ngp-1/llvm-project/clang-ci/libcxx/test/libcxx/gdb/gdb_pretty_printer_test.py", line 123, in <module>
|     gdb.execute("run")
| gdb.error: Warning:
| Cannot insert breakpoint 1.
| Cannot access memory at address 0x55555555637c
```

I thought that maybe the problem is that, the program to be debugged by `dbg` does not compile with the compiler that includes my changes. I decided to just compile the test file with my modified compiler. I copied the test file. Removed one include that is only needed for manual debugging and recreated the command used in the CI as much as I could. The command looks like this now:
```
/home/spaits/repo/llvm-project/build/bin/clang++ -pthread --target=x86_64-unknown-linux-gnu /home/spaits/cpp/deb.cpp -o /home/spaits/cpp/deb.out -std=c++26 -Werror -Wall -Wctad-maybe-unsupported -Wextra -Wshadow -Wundef -Wunused-template -Wno-unused-command-line-argument -Wno-attributes -Wno-pessimizing-move -Wno-noexcept-type -Wno-atomic-alignment -Wno-reserved-module-identifier -Wdeprecated-copy -Wdeprecated-copy-dtor -Wno-user-defined-literals -Wno-tautological-compare -Wsign-compare -Wunused-variable -Wunused-parameter -Wunreachable-code -Wno-unused-local-typedef -Wno-local-type-template-args -Wno-c++11-extensions -Wno-unknown-pragmas -Wno-pass-failed -Wno-mismatched-new-delete -Wno-redundant-move -Wno-self-move -D_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER -D_LIBCPP_ENABLE_EXPERIMENTAL -D_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_NONE -Werror=thread-safety -Wuser-defined-warnings -g   -lc++ -latomic
0 spaits@work-laptop:~/cpp$ 
```
Because of the GCC 13+ prereq. I did not build libc++ for my self, but used the one available on my machine.
This way the file has compiled successfully.

I will still try to run the tests myself.

Until that if you have any idea what should I, do what wen possibly wrong pleas share that with me.

Thanks.

--- a/clang/lib/Sema/SemaTemplateDeduction.cpp
+++ b/clang/lib/Sema/SemaTemplateDeduction.cpp
@@ -861,14 +861,10 @@ public:
 
   // Return the size of the saved packs if all of them has the same size.
   std::optional<unsigned> getSavedPackSizeIfAllEqual() const {
-    if (Packs.size() == 0 ||
-        Packs[0].Saved.getKind() != clang::TemplateArgument::Pack)
-      return {};
     unsigned PackSize = Packs[0].Saved.pack_size();
 
     if (std::all_of(Packs.begin() + 1, Packs.end(), [&PackSize](const auto &P) {
-          return P.Saved.getKind() == TemplateArgument::Pack &&
-                 P.Saved.pack_size() == PackSize;
+          return P.Saved.pack_size() == PackSize;
         }))
       return PackSize;
     return {};

--- a/clang/lib/Sema/SemaTemplateDeduction.cpp
+++ b/clang/lib/Sema/SemaTemplateDeduction.cpp
@@ -4406,7 +4406,7 @@ Sema::TemplateDeductionResult Sema::DeduceTemplateArguments(
         // previously we substitute the pack here in a similar fashion as
         // above with the trailing parameter packs. The main difference here is
         // that, in this case we are not processing all of the remaining
-        // arguments. We are only process as many arguments as much we have in
+        // arguments. We are only process as many arguments as we have in
         // the already deduced parameter.
         std::optional<unsigned> ArgPosAfterSubstitution =
             PackScope.getSavedPackSizeIfAllEqual();

--- a/clang/test/SemaTemplate/deduction.cpp
+++ b/clang/test/SemaTemplate/deduction.cpp
@@ -454,6 +454,12 @@ namespace deduction_after_explicit_pack {
     foo(args_tag<const int,const int, const int>{}, args_tag<int, int, int>{}, 4, 8, 9, 15, 16, 23, 1);
   }
 
+  template <typename... Y, typename... T>
+  void foo2(args_tag<Y...>, args_tag<T...>, type_identity_t<T>..., type_identity_t<T>...) {}
+  void call_foo2() {
+    foo2(args_tag<const int,const int, const int>{}, args_tag<int, int, int>{}, 4, 8, 9, 15, 16, 23);
+  }
+
   template <typename... Y, typename... T> void baz(args_tag<T...>, T..., T...) {}
   void call_baz() {
     baz(args_tag<int, int>{}, 1, 2, 3, 4);

